### PR TITLE
feat: add the repository mock and also delete and update functions

### PR DESCRIPTION
**Pull request by GuiGuerreiroo**

### CODE REVIEW

File: src/shared/domain/repositories/user_repository_mock.py
Comment (LucasCrapino) [inline src/shared/domain/repositories/user_repository_mock.py:1]: eu adicionaria mais users mockados, um com estado rejected, outro com role de presidente, e também de outra entidade, só por precaução pra gente usar mais pra frente

Comment (GuiGuerreiroo) [reply src/shared/domain/repositories/user_repository_mock.py:1]: perfeito vou fazer sim, brigado pelo feedback

File: src/shared/infra/repositories/user_repository_mock.py [hunk 1]
```diff
@@ -1,51 +1,57 @@
...
-        self.user_counter = 3
-
+    
     def get_user(self, user_id: int) -> User:
         for user in self.users:
             if user.user_id == user_id:
                 return user
-        raise NoItemsFound("user_id")
-
+        raise NoItemsFound(user_id)
```
Comment (lseixas) [inline src/shared/infra/repositories/user_repository_mock.py:27]: boa

File: src/shared/domain/repositories/user_repository_interface.py [hunk 2]
```diff
@@ -12,7 +16,7 @@ def get_user(self, user_id: int) -> User:
         If user not found raise NoItemsFound
         """
         pass
-
+        
     @abstractmethod
     def get_all_user(self) -> List[User]:
         pass
```
Comment (lseixas) [inline src/shared/domain/repositories/user_repository_interface.py]: adiciona uma doc_string aqui igual tem nos outros métodos @Pistori1

File: src/shared/domain/repositories/user_repository_interface.py
Comment (lseixas) [inline src/shared/domain/repositories/user_repository_interface.py:1]: adiciona docstring nesses métodos que estao faltando, pf @Pistori1

File: src/shared/infra/repositories/user_repository_mock.py [hunk 3]
```diff
@@ -1,51 +1,57 @@
...
-
+    
     def create_user(self, new_user: User) -> User:
         self.users.append(new_user)
-        self.user_counter += 1
         return new_user
 
-    def delete_user(self, user_id: int) -> User:
-        for idx, user in enumerate(self.users):
+    def delete_user(self, user_id: str):
```
Comment (lseixas) [inline src/shared/infra/repositories/user_repository_mock.py:38]: brabissimo

File: src/shared/infra/repositories/user_repository_mock.py [hunk 4]
```diff
@@ -1,51 +1,57 @@
...
+                if new_state != None:
+                    user.state= new_state
+                if new_role != None:
+                    user.role= new_role
+                if new_course != None:
+                    user.course= new_course
+                if new_year != None:
+                    user.year= new_year
+                if new_organization != None:
+                    user.organization= new_organization
```
Comment (lseixas) [inline src/shared/infra/repositories/user_repository_mock.py:57]: boa

File: tests/shared/infra/repositories/test_user_repository_mock.py [hunk 5]
```diff
@@ -1,72 +1,113 @@
...
+        assert user.email == "25.00178-5@maua.br"
+        assert user.user_id == "550e8400-e29b-41d4-a716-446655440000"
+        assert user.state == STATE.PENDING
+        assert user.role == ROLE.USER
+
+
+    def test_get_user_not_found(self):
+        repo = UserRepositoryMock()
+        with pytest.raises(NoItemsFound):
+            user = repo.get_user(69)
```
Comment (lseixas) [inline tests/shared/infra/repositories/test_user_repository_mock.py:28]: top

File: tests/shared/infra/repositories/test_user_repository_mock.py [hunk 6]
```diff
@@ -1,72 +1,113 @@
...
+        assert deleted_user.email == "25.00178-5@maua.br"
+        assert deleted_user.user_id == "550e8400-e29b-41d4-a716-446655440000"
+        assert deleted_user.state == STATE.PENDING
+        assert deleted_user.role == ROLE.USER
+
+
+    def test_delete_user_not_found(self):
+        repo = UserRepositoryMock()
+        with pytest.raises(NoItemsFound):
+            deleted_user = repo.delete_user(user_id="550e8400-e29b-41d4-a716-446655440005")
```
Comment (lseixas) [inline tests/shared/infra/repositories/test_user_repository_mock.py:73]: boa

File: tests/shared/infra/repositories/test_user_repository_mock.py [hunk 7]
```diff
@@ -1,72 +1,113 @@
...
+        assert repo.users[1].role == ROLE.ADM
+        assert repo.users[1].course == COURSE.CIC
+        assert repo.users[1].year == 4
+        assert repo.users[1].organization == ORGANIZATION.DEV
+
+    def test_update_user_not_found(self):
+        repo = UserRepositoryMock()
+
+        with pytest.raises(NoItemsFound):
+            updated_user= repo.update_user(user_id="550e8400-e29b-41d4-a716-446655440005", new_state=STATE.APPROVED, new_role=ROLE.ADM, new_course=COURSE.CIC, new_year=4, new_organization=ORGANIZATION.DEV)
```
Comment (lseixas) [inline tests/shared/infra/repositories/test_user_repository_mock.py:111]: brabo, sempre muito importante cobrir esses edge cases ou caso dos erros, muito bom

**Review by lseixas — CHANGES_REQUESTED** [review]
@Pistori1 só coloca as docstring dos métodos da interface e ai eu aprovo, coisa rápida

File: src/shared/domain/repositories/user_repository_mock.py
Comment (lseixas) [reply src/shared/domain/repositories/user_repository_mock.py:1]: boa rpz!

File: src/shared/domain/entities/user.py
Comment (lseixas) [inline src/shared/domain/entities/user.py:1]: @LucasCrapino mano, acho que aqui pode deixar como o Guerreiro fez. Pensamos na seguinte logica: No login do usuário, receberemos apenas essas informações: user_id, email, nome. A partir dessa informações conseguimos deduzir apenas o ra, ou seja, os outros parâmetros que serão passados quando o usuário for CRIADO, terão que ser None por padrão. Por exemplo: Apenas com essas informações, nao conseguimos saber qual curso, organização ou ano a pessoa esta, concorda? Entao, na parte das validações o ideal é permitir que esses parâmetros sejam passados como None. Além disso, no init creio que seja melhor por uma questão de leiturabilidade não permitirmos default values como None, creio que é melhor no geral indicarmos explicitamente, quando estamos criando uma entidade usuário na rota, que certos parâmetros foram passados como None. Um exemplo disso para ilustrar seria: User(user_id="user_id", email="email"..... year=None, Org=None) é preferível, na minha opinião, do que um: User(user_id="user_id", email="email"....), sem declarar explicitamente que esses parâmetros sao None, entende? Podemos discutir isso na daily depois @LucasCrapino ?

File: src/shared/domain/enums/organization_enum.py
Comment (lseixas) [inline src/shared/domain/enums/organization_enum.py:1]: boa

**Review by lseixas — APPROVED** [review]
